{
  if (!async) {
    try {
      writeToFile(path,content,size);
    }
 catch (    IOException e) {
      LOG.error(e);
      return;
    }
  }
  taskProcessor.add(new ThrowableRunnable<Exception>(){
    @Override public void run() throws Exception {
      if (async) {
        writeToFile(path,content,size);
      }
      doAdd(path);
    }
  }
);
}
